src/internal_events/http_client.rs

```diff
@@ -109,3 +109,29 @@ impl<'a, B: HttpBody> std::fmt::Display for FormatBody<'a, B> {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct HttpBodyDataReceived<'a> {
+    pub buf: &'a mut Vec<u8>,
+}
+
+impl<'a> InternalEvent for HttpBodyDataReceived<'a> {
+    fn emit(self) {
+        // Check if the buffer is empty before any operation
+        if self.buf.is_empty() {
+            return;
+        }
+
+        // Truncate the buffer to the first 20 elements if the buffer is over 20 elements
+        // TODO: make this configurable
+        if self.buf.len() > 20 {
+            self.buf.truncate(20);
+        }
```

Comment:
Since this limit is also the threshold after which to issue the event, this should be handled fully in the caller.